### PR TITLE
clang-tidy fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by wilx** [comment]
Thank you for your contribution.

I have integrated most of your change sets except these: cbbc3ed2628ad2795a0b07b4bda48efaf7918e35, b2c9052d85d9b1b8dc63b6b7ddfa5d9407aa498c, e8da854f165627c33037586c52de0a3f93760353.

The "use make_unique" change set needs to first be fixed. See my review comment. Also, I honestly don't see much value in it but I will accept it if you fix  the header issue.

I am not going to accept the pass by value changeset. It replaces only some of the reference parameters by value parameter and it makes the interfaces ugly, IMO.

I am also not going to accept the "remove redundant static" changeset. I agree with you that it would be nice not to have to have the static keyword there but it is there for a reason. The reason is that Cygwin's compiler/linker exports all symbols including the anonymous namespace symbols when we build the library. The `static` keyword is a remedy, functions marked such do not appear in DLL exports. Or at least it used to be like this, I haven't checked for years now. I would need a positive confirmation that current Cygwin does not export anonymous namespace functions to accept the change set.

**Comment by neheb** [comment]
I don't see the review comment anywhere.

The static thing I can test. I have an up to date Cygwin install.

**Comment by wilx** [comment]
I forgot to hit Submit review. It should be visible now.

File: src/log4judpappender.cxx [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include <cwctype>
 #else
 #include <cctype>
+#include <memory>
```
[the remark this answers is not in the document]
Comment (neheb) [reply src/log4judpappender.cxx]: clang-tidy did this. Will fix.

**Comment by neheb** [comment]
https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html says it's a compiler optimization. I've no real opinion on it.

**Pull request closed without merging**